packages/legacy-client/src/client/objectSets/OsdkObjectSet.ts

```diff
@@ -0,0 +1,138 @@
+import type { OsdkLegacyObjectFrom } from "../OsdkObject";
+import { createObjectSetAggregationStep } from "./createObjectSetAggregationStep";
+import { createObjectSetOrderByStep } from "./createObjectSetOrderByStep";
+import { createObjectSetSearchAround } from "./createObjectSetSearchAround";
+import { createObjectSetTerminalLoadStep } from "./createObjectSetTerminalLoadStep";
+
+export function createOsdkObjectSet<
+  O extends OntologyDefinition<any>,
+  K extends ObjectTypesFrom<O>,
+>(
+  clientContext: ClientContext,
+  objectSetDefinition: ObjectSetDefinition,
+  ontologyDefinition: O,
+): ObjectSet<OsdkLegacyObjectFrom<O, K>> {
+  const objectSet: ObjectSetOperations<OsdkLegacyObjectFrom<O, K>> = {
+    union(...otherObjectSets): ObjectSet<OsdkLegacyObjectFrom<O, K>> {
+      return createOsdkObjectSet(
+        clientContext,
+        {
+          type: "union",
+          objectSets: [
+            objectSetDefinition,
+            ...otherObjectSets.map((s) => s.definition),
+          ],
+        },
+        ontologyDefinition,
+      );
+    },
+    intersect(...otherObjectSets): ObjectSet<OsdkLegacyObjectFrom<O, K>> {
+      return createOsdkObjectSet(
+        clientContext,
+        {
+          type: "intersect",
+          objectSets: [
+            objectSetDefinition,
+            ...otherObjectSets.map((s) => s.definition),
+          ],
+        },
+        ontologyDefinition,
+      );
+    },
+    subtract(...otherObjectSets): ObjectSet<OsdkLegacyObjectFrom<O, K>> {
+      return createOsdkObjectSet(
+        clientContext,
+        {
+          type: "subtract",
+          objectSets: [
+            objectSetDefinition,
+            ...otherObjectSets.map((s) => s.definition),
+          ],
+        },
+        ontologyDefinition,
+      );
+    },
+    where(predicate): ObjectSet<OsdkLegacyObjectFrom<O, K>> {
+      throw new Error("not implemented");
+    },
+    select<T extends keyof SelectableProperties<OsdkLegacyObjectFrom<O, K>>>(
+      properties: T[],
+    ): FilteredPropertiesTerminalOperations<OsdkLegacyObjectFrom<O, K>, T[]> {
+      throw new Error("not implemented");
+    },
+  };
+
+  return {
+    definition: objectSetDefinition,
+    ...objectSet,
+    ...createObjectSetSearchAround<O, K>(),
+    ...createObjectSetTerminalLoadStep<O, K>(),
+    ...createObjectSetOrderByStep<O, K>(),
+    ...createObjectSetAggregationStep<O, K>(),
+  };
+}
+
+export function createBaseOsdkObjectSet<
+  O extends OntologyDefinition<any>,
+  K extends ObjectTypesFrom<O>,
+>(
+  clientContext: ClientContext,
+  apiName: string,
+  ontologyDefinition: O,
+): BaseObjectSet<OsdkLegacyObjectFrom<O, K>> {
+  const baseObjectSetDefinition: BaseObjectSetDefinition = {
+    type: "base",
+    objectType: apiName,
+  };
+
+  const objectSet: BaseObjectSetOperations<OsdkLegacyObjectFrom<O, K>> = {
+    apiName: apiName as OsdkLegacyObjectFrom<O, K>["__apiName"],
+    description: "test",
```

Comment:
Sneaky thing that should get fixed before release but isn't a `throw new Error()`.

Reply:
Added a new description property that we can use on the definition